### ISSUE TITLE
voluntary exits 'no peers on channel'

### ISSUE THREAD

**Issue by rolfyone** [Consensys-Incorporated/teku#8158]
When submitting a voluntary exit in deneb, 'no peers on topic' has been reported.

We need to investigate this, the Deneb spec says `The voluntary_exit topic is implicitly modified due to the lock-in use of CAPELLA_FORK_VERSION for this message signature validation for EIP-7044.`

This possibly means that the voluntary exit digest should be locked at the capella digest, so we're essentially on the wrong gossip topic to find peers for exits.

**Comment by lucassaldanha** [comment]
I haven't got confirmation from the other client teams if we should fix the fork digest, but I checked a Lighthouse node in Sepolia and all topics are using an updated fork digest after Deneb, and all topics match the topics that our Teku nodes in Sepolia are subscribed to.

So either we do not need to fix the fork digest for the `voluntary_exit` topic, or both Teku and LH are doing something wrong :)

**Comment by zilm13** [comment]
### Spec:


The voluntary_exit topic is implicitly modified due to the lock-in use of CAPELLA_FORK_VERSION for this message signature validation for EIP-7044. [spec](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/p2p-interface.md#topics-and-messages)

> I read it that topic is Capella fixed.

```
The specific makes one change to the state transition function:

    Modify `process_voluntary_exit` to compute the signing domain and root fixed on `CAPELLA_FORK_VERSION`.

Additionally, the `voluntary_exit` gossip conditions are implicitly modified to support this change.
```
[EIP-7044](https://eips.ethereum.org/EIPS/eip-7044), which is the part of Deneb
> This could be read in two ways: either topic is Capella-fixed or topic is changed according to the fork, but validation should not reject Capella-signed voluntary exits. 

### Our code:
[Validation lock-in to Capella](https://github.com/Consensys/teku/blob/78a929fa0bbe76eaac8d2bdd3dcea9d9125db28f/ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/deneb/helpers/BeaconStateAccessorsDeneb.java#L43-L52)

Create Voluntary Exits (subcommand) -  no lock-in: [[1]](https://github.com/Consensys/teku/blob/7e3fa5dc4e4d8ad03c566eabdda9884f7f0323c0/teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java#L301), [[2]](https://github.com/Consensys/teku/blob/7e3fa5dc4e4d8ad03c566eabdda9884f7f0323c0/teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java#L240-L247)

Gossip subscriptions - [no lock-in](https://github.com/Consensys/teku/blob/f31b239f7788f07d8e9c2b6e7544c5e16b01a259/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/forks/versions/GossipForkSubscriptionsPhase0.java#L157-L164)

**Comment by zilm13** [comment]
Issue opened for topic clarification https://github.com/ethereum/consensus-specs/issues/3638 as we are not sure on this

**Comment by zilm13** [comment]
`Create Voluntary Exits (subcommand) - no lock-in: ` - not true
Actually we do this deep in spec beacon state accessors. Just added more tests to avoid confusion on this side.

**Comment by rolfyone** [comment]
```Not publishing voluntary exit because no gossip subscriptions are active for slot xxxx```

**Comment by zilm13** [comment]
Fixed with #8205
